### PR TITLE
kent-tools 313 (new formula)

### PR DESCRIPTION
**Pull request by jonchang**
UCSC Kent Tools, second try. Previous PR was #37.

Since [ucsc-genome-browser](https://github.com/Homebrew/homebrew-science/blob/master/ucsc-genome-browser.rb) is now keg only, these standalone tools were requested by #1904. Tested on CentOS 5.6, Mac OS X 10.6.8 and 10.10.1.

### CODE REVIEW

**Comment by sjackman** [comment]
[Log](http://bot.brew.sh/job/Homebrew%20Science%20Pull%20Requests/1251/version=yosemite/console)

```
==> brew audit Homebrew/science/kent-tools --strict			 FAILED
==> brew style kent-tools
== /usr/local/Library/Taps/homebrew/homebrew-science/kent-tools.rb ==
C:  2: 12: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C:  3:  7: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C:  4:  8: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C:  5: 10: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C:  8: 14: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C:  9: 14: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C: 12: 22: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C: 13: 21: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C: 15: 13: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C: 25: 15: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C: 28: 22: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C: 28: 30: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C: 29: 14: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C: 44: 20: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C: 49: 20: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C: 49: 31: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

1 file inspected, 16 offenses detected
```

File: kent-tools.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      http://hgdownload.cse.ucsc.edu/admin/exe/
+
+    If you need the full UCSC Genome Browser, run:
+      brew install ucsc-genome-browser
+    EOS
+  end
+
+  test do
+    mktemp do
+      Pathname.new('test.fa').write <<-EOF.undent
```
Comment (iml) [inline kent-tools.rb]: `mktemp` and `Pathname` are unnecessary. Just use `(testpath/"test.fa").write`

**Comment by jonchang** [comment]
Pushed up fixes as fbf8b1f.

```
 $ brew audit --strict kent-tools.rb
==> brew style kent-tools

1 file inspected, no offenses detected
```

**Comment by sjackman** [comment]
`kent-tools` conflicts with `blat`. Please remove the files that conflict with `blat`.

The utility `bin/calc` also conflicts with the formula `calc`. If it's not necessary to `kent-tools`, I'd suggest removing it or renaming it to `kent-tools-calc`.

```
❯❯❯ brew link --overwrite --dry-run kent-tools
Would remove:
/usr/local/bin/blat -> /usr/local/Cellar/blat/35/bin/blat
/usr/local/bin/calc -> /usr/local/Cellar/calc/2.12.5.0/bin/calc
/usr/local/bin/faToTwoBit -> /usr/local/Cellar/blat/35/bin/faToTwoBit
/usr/local/bin/gfClient -> /usr/local/Cellar/blat/35/bin/gfClient
/usr/local/bin/gfServer -> /usr/local/Cellar/blat/35/bin/gfServer
/usr/local/bin/nibFrag -> /usr/local/Cellar/blat/35/bin/nibFrag
/usr/local/bin/pslPretty -> /usr/local/Cellar/blat/35/bin/pslPretty
/usr/local/bin/pslReps -> /usr/local/Cellar/blat/35/bin/pslReps
/usr/local/bin/pslSort -> /usr/local/Cellar/blat/35/bin/pslSort
/usr/local/bin/twoBitInfo -> /usr/local/Cellar/blat/35/bin/twoBitInfo
/usr/local/bin/twoBitToFa -> /usr/local/Cellar/blat/35/bin/twoBitToFa
```

**Comment by jonchang** [comment]
Pushed requested fixes as 6e6f7f4. Also upgraded to v313.

**Comment by sjackman** [comment]
Thanks for your contribution, Jon. MySQL is a big-ish dependency for someone who wants only the FASTA scripts. Should it instead be a recommended or optional dependency?

**Comment by sjackman** [comment]
```
~ ❯❯❯ brew fetch kent-tools
==> Fetching kent-tools from homebrew/homebrew-science
==> Downloading https://homebrew.bintray.com/bottles-science/kent-tools-313.yosemite.bottle.tar.gz

curl: (22) The requested URL returned error: 401 Unauthorized
Error: Failed to download resource "kent-tools"
Download failed: https://homebrew.bintray.com/bottles-science/kent-tools-313.yosemite.bottle.tar.gz
```

**Comment by sjackman** [comment]
@BrewTestBot test this please

**Comment by sjackman** [comment]
The 313 bottles didn't seem to get uploaded correctly. https://bintray.com/homebrew/bottles-science/kent-tools/view#files

**Comment by jonchang** [comment]
@sjackman I suspect that most (or all) of these tools don't actually need mysql, but the linker just links them anyway. I haven't investigated enough though.

**Comment by iml** [comment]
@sjackman I reset back a couple of commits (locally) and reran `brew pull --bottle` and it seems to have published the bottles properly.

**Comment by sjackman** [comment]
Thanks, Ian.

**Pull request closed without merging**